crates/compiler/collections/src/vec_map.rs

```diff
@@ -84,6 +84,13 @@ impl<K: PartialEq, V> VecMap<K, V> {
         }
     }
 
+    pub fn get_with_index(&self, key: &K) -> Option<(usize, &V)> {
+        self.keys
+            .iter()
+            .position(|x| x == key)
+            .map(|index| (index, &self.values[index]))
+    }
```

Comment:
I was trying to prevent scanning the `params` map unnecessarily, but I'm happy to go with that approach if you think that's ok.

Reply:
Maybe I should have `ids` and `names` `Vec`s in `ScopeModules` directly, instead of using `VecMap`. That way, I could have access to the index for lookups without making `VecMap` leaky.

Reply:
I think either alternative is fine. Indexing the map on the ModuleId seems reasonable too.

Reply:
Done